rules/library.bzl

```diff
@@ -664,17 +664,31 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
         lib_names.append(cpp_libname)
 
     additional_objc_copts.append("-I.")
+    index_while_building_objc_copts = select({
+        "@build_bazel_rules_ios//:index_while_building_v2": [
+            # Note: this won't work work for remote caching yet. It uses a
+            # _different_ global index for objc than so that the BEP grep in
+            # rules_ios picks this up.
+            # Checkout the task roadmap for future improvements:
+            # Docs/index_while_building.md
+            "-index-store-path",
+            "bazel-out/rules_ios_global_index_store.indexstore",
```

Comment:
> additionally, does the bumped version of index-import alleviate the issues we saw of re-mapping being done per-indexstore as opopsed to per-unit? that was the original motivation for separating them out.

It looks like Jerry's passing the `-incremental` flag to index-import which *should* alleviate the problems we saw with using a global objc indexstore.

Jerry, could you make sure to test out a build which compiles a lot of obj-c files? Ensure that index-import runs fast on rebuilds of those projects (because on rebuilds, index-import should not be moving or remapping any unit files at all.)

Reply:
It should point at `bazel-out` to make sure there is a singe index store.

@amberdixon until we implement `Local optimization - sans index-import`, it still has to remap https://github.com/bazel-ios/rules_ios/blob/master/docs/index_while_building.md#local-optimization---sans-index-import

We should have dropped atleast an order of magnitude of index data it needs to process with this PR though

Reply:
This is probably another change that we'll want for indexstore size: https://github.com/bazelbuild/rules_swift/issues/575 - I can propose that upstream or we can add it into rules_ios